### DISCUSSION TITLE
Save the fitted model

### DISCUSSION THREAD

**Discussion by vnikoofard** [MilesCranmer/PySR#119] [Q&A] [answered]
Hi,

Is there any way to save the fitted model? I tried to save as pickle by dumping 
```
with open('sr_model.pk', 'wb') as sr_model_file:
    pickle.dump(sr_model, sr_model_file)
```
but the following message was thrown:
```
[PicklingError: Can't pickle <function _lambdifygenerated at 0x7fd3ec06aee0>: attribute lookup _lambdifygenerated on __main__ failed]()
```

**Comment by MilesCranmer** [comment]
Good question. I have never tried this, but note that PySR saves the equations to a csv file during the search. In the PySRRegressor model, this is set by `equation_file` (or is chosen automatically by the model).

So what you could do is get the equation file (`sr_model.equation_file`), then create a new PySRRegressor with the same settings, then run `sr_model.get_hof()`. This should load the equations from the csv file.

Let me know if this works!

**Comment by MilesCranmer** [comment]
One other option: use `dill` https://github.com/uqfoundation/dill instead of pickle:
```python
# import pickle as pkl
import dill as pkl
```
This natively supports lambda functions, so it might work out of the box!

(You might have to set `sr_model.raw_julia_state = None` before saving the model, however, as the `raw_julia_state` contains some ctypes pointers, which I don't think even `dill` would support.)

**Reply by vnikoofard** [reply]
Hi,

Still I am not tried the first idea but the `dill` pickling throw the following error, even with `sr_model.raw_julia_state=None`

```
[PicklingError: Can't pickle inv: it's not found as __main__.inv]()
```
Maybe this error be related to the unary operator `"inv(x) = 1/x",`. I will try without it.

**Comment by vnikoofard** [comment]
No, still the `dill` can't save the model.

**Comment by MilesCranmer** [comment, marked as answer]
Hi @vnikoofard,

With v0.9.0 of PySR (https://github.com/MilesCranmer/PySR/pull/146), you can now pickle and unpickle the PySRRegressor without issue.

Best,
Miles

**Reply by vnikoofard** [reply]
Great! Thanks for making this excellent software available for all the community.
